### PR TITLE
Optimize CI a bit

### PR DESCRIPTION
**Pull request by Goooler**

### CODE REVIEW

File: .github/workflows/build-unsigned.yaml [hunk 1]
```diff
@@ -27,22 +27,19 @@ jobs:
...
       - name: Setup Go
-        uses: actions/setup-go@v2
+        uses: actions/setup-go@v3
         with:
           go-version: 1.17
       - name: Build
-        run: ./gradlew --no-daemon app:assembleFossRelease
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: --no-daemon app:assembleFossRelease
```
Comment (Goooler) [inline .github/workflows/build-unsigned.yaml:45]: https://github.com/gradle/gradle-build-action

Ref https://github.com/detekt/detekt/pull/3723, it's really so fast to run jobs, and many other advantages.

File: .github/workflows/build-unsigned.yaml [hunk 2]
```diff
@@ -27,22 +27,19 @@ jobs:
...
         with:
-          distribution: 'adopt'
-          java-version: 11
-      - name: Gradle cache
-        uses: actions/cache@v2
-        with:
-          path: ~/.gradle/caches
-          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle/wrapper/gradle-wrapper.properties', '**/buildSrc/src/main/kotlin/**.kt') }}
+          distribution: 'zulu'
+          java-version: 17
```
Comment (Goooler) [inline .github/workflows/build-unsigned.yaml:37]: Java 17, faster.

File: .github/workflows/build-unsigned.yaml [hunk 3]
```diff
@@ -27,22 +27,19 @@ jobs:
...
       - name: Setup Go
-        uses: actions/setup-go@v2
+        uses: actions/setup-go@v3
         with:
           go-version: 1.17
```
Comment (Kr328) [inline .github/workflows/build-unsigned.yaml:41]: Bump to go 1.18 please

**Review by Kr328 — CHANGES_REQUESTED** [review]

File: .github/workflows/build-unsigned.yaml (hunk 3 above)
Comment (Goooler) [inline .github/workflows/build-unsigned.yaml:41]:
```suggestion
          go-version: 1.18
```

Comment (Goooler) [reply .github/workflows/build-unsigned.yaml]: 32610a4f490eba070cc2a0738eb1dfe05c0c70ee

**Pull request merged**